cli/pkg/k8s/k8s.go

```
@@ -518,4 +526,239 @@ func DeletePods(namespace string, labels map[string]string) error {
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to create certificate: %v", err)
+	}
+
+	certOut := new(bytes.Buffer)
+	if err := pem.Encode(certOut, &pem.Block{Type: "CERTIFICATE", Bytes: derBytes}); err != nil {
+		return nil, nil, fmt.Errorf("failed to write data to cert.pem: %v", err)
+	}
+
+	privBytes, err := x509.MarshalPKCS8PrivateKey(priv)
+	if err != nil {
+		return nil, nil, fmt.Errorf("unable to marshal private key: %v", err)
+	}
+
+	keyOut := new(bytes.Buffer)
+	if err := pem.Encode(keyOut, &pem.Block{Type: "PRIVATE KEY", Bytes: privBytes}); err != nil {
+		return nil, nil, fmt.Errorf("failed to encode private key: %v", err)
+	}
+
+	return certOut.Bytes(), keyOut.Bytes(), nil
+}
+
+func publicKey(priv interface{}) interface{} {
+	switch k := priv.(type) {
+	case *rsa.PrivateKey:
+		return &k.PublicKey
+	default:
+		return nil
+	}
+}
+
+func checkSecretExists(secretName, namespace string) (bool, error) {
+	client, _ := kubeInit()
+	secExists, err := client.CoreV1().Secrets(namespace).Get(context.Background(), secretName, metav1.GetOptions{})
+	e1 := errors.New("secrets \"" + secretName + "\" not found")
+	if err != nil && err.Error() != e1.Error() {
+		return false, fmt.Errorf("failed to get %s secret present or not %v", secretName, err)
+	}
+
+	if secExists.Name != "" {
+		fmt.Printf("using existing %s secret\n", secretName)
+		return true, nil
+	}
+
+	return false, nil
+}
+
+func DeleteTimescaleDBSecrets(releaseName, namespace string, backUpEnabled bool) {
+	client, _ := kubeInit()
+	fmt.Println("Deleting TimescaleDB secrets...")
+	credentialsSecret := []string{releaseName + "-credentials", releaseName + "-certificate"}
+	if backUpEnabled {
+		credentialsSecret = append(credentialsSecret, releaseName+"-pgbackrest")
+	}
+	for _, s := range credentialsSecret {
+		err := client.CoreV1().Secrets(namespace).Delete(context.Background(), s, metav1.DeleteOptions{})
+		if err != nil {
```

let's check https://pkg.go.dev/k8s.io/apimachinery/pkg/api/errors#IsNotFound and ignore error if not found